HeavyFlavorAnalysis/Onia2MuMu/src/OniaPhotonConversionProducer.cc

```diff
@@ -140,63 +139,63 @@ void OniaPhotonConversionProducer::produce(edm::Event& event, const edm::EventSe
+
+     bool flag1 = true;
+     bool flag2 = true;
+     bool flag3 = true;
+     bool flag4 = true;
 
     // The logic implies that by default this flags are true and if the check are not wanted conversions are saved.
     // If checks are required and failed then don't save the conversion.
 
-    bool flagTkVtxCompatibility  = true;
-    if (! checkTkVtxCompatibility(*conv,*priVtxs.product())) {
+     bool flagTkVtxCompatibility  = true;
+     if (! checkTkVtxCompatibility(*conv,*priVtxs.product())) {
        flagTkVtxCompatibility = false;
        if (wantTkVtxCompatibility_) {
           TkVtxC++;
           flag1 = false;
        }
-    }
-    bool flagCompatibleInnerHits = false;
-    if (conv->tracks().size()==2) {
+     }
+     bool flagCompatibleInnerHits = false;
+     if (conv->tracks().size()==2) {
        reco::HitPattern hitPatA=conv->tracks().at(0)->hitPattern();
        reco::HitPattern hitPatB=conv->tracks().at(1)->hitPattern();
        if ( foundCompatibleInnerHits(hitPatA,hitPatB) && foundCompatibleInnerHits(hitPatB,hitPatA) ) flagCompatibleInnerHits = true;
-    }
-    if (wantCompatibleInnerHits_ && ! flagCompatibleInnerHits) {
+     }
+     if (wantCompatibleInnerHits_ && ! flagCompatibleInnerHits) {
        CInnerHits++;
        flag2 = false;
-    }
-    bool flagHighpurity = true;
-    if (!HighpuritySubset(*conv,*priVtxs.product())) {
+     }
+     bool flagHighpurity = true;
+     if (!HighpuritySubset(*conv,*priVtxs.product())) {
        flagHighpurity = false;
        if (wantHighpurity_) {
           highpurity_count++;
           flag3 = false;
        }
-    }
-    bool pizero_rejected = false;
-    bool large_pizero_window = CheckPi0(*conv, pfphotons, pizero_rejected);
-    if (pi0OnlineSwitch_ && pizero_rejected) {
-      pizero_fail++;
-      flag4 = false;
-    } 
+     }
+     bool pizero_rejected = false;
+     bool large_pizero_window = CheckPi0(*conv, pfphotons, pizero_rejected);
+     if (pi0OnlineSwitch_ && pizero_rejected) {
+       pizero_fail++;
+       flag4 = false;
+     }
+
+     int flags = 0;
+     if (flag1 && flag2 && flag3 && flag4){
+        flags = PackFlags(*conv,flagTkVtxCompatibility,flagCompatibleInnerHits,flagHighpurity,pizero_rejected,large_pizero_window);
+        final_conversion++;
+     }
 
-    if (flag1 && flag2 && flag3 && flag4){
-        int flags = PackFlags(*conv,flagTkVtxCompatibility,flagCompatibleInnerHits,flagHighpurity,pizero_rejected,large_pizero_window);
-        flagCollection.push_back(flags);
-       	outCollection->push_back(*conv);
-	final_conversion++;
-    }
-  }
-  removeDuplicates(*outCollection,flagCollection);
-  
-  int i = -1; 
-  for (reco::ConversionCollection::const_iterator conv = outCollection->begin(); conv != outCollection->end(); ++conv){
-     i++;
      pat::CompositeCandidate *pat_conv = makePhotonCandidate(*conv);
-     pat_conv->addUserInt("flags",flagCollection.at(i));
+     pat_conv->addUserInt("flags",flags);
```

Comment:
line 198 & 199 should be in the if-clause starting at line 192 to have the same behavior as before this pull request.